packages/core/src/components/image-overlay/image-overlay-overlay.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Prop,
+} from "@stencil/core";
+import * as focusTrap from "focus-trap";
+
+const margin = 16;
+
+@Component({
```

Comment:
Is dit niet een beetje overkill? Twee Web Componenten?

Als je code wilt splitten, heb je al eens naar Functional Components gekeken?